web/frontend/src/components/arm/index.svelte

```diff
@@ -85,7 +87,7 @@ let modifyAllStatus: ArmStatus = {
   joint_pieces: [],
 };
 
-const armClient = new ArmClient(client, name, { requestLogger: rcLogConditionally });
+const armClient = new ArmClient($client, name, { requestLogger: rcLogConditionally });
```

Comment:
Svelte neophyte question: how does this work with reactivity? Will Svelte re-create `armClient` if the underlying `client` is replaced?

Why is it not something like this?

```svelte
$: armClient = new ArmClient($client, ...)
```

Follow-up question: should there be a `useArmClient` hook instead of doing this here?

Reply:
Yeah this is a good call out. I originally left as is since client is unchanged after app initialization, but might be good to guard against that possibly not being the case in the future. This is again related to problems with not knowing if client should be a persistent object that is only generated once at the beginning of the app lifecycle - and if it is it could just be made a var returned by `useClient` and not a store - or if it will be generated multiple times during app lifecycle.

> Follow-up question: should there be a useArmClient hook instead of doing this here?

Definitely. I might want to leave these two problems out since this PR is already big